src/domain/entities/figma-design-identifier.ts

```diff
@@ -40,12 +40,16 @@ export class FigmaDesignIdentifier {
 		);
 
 		const fileKey = pathComponents[filePathComponentId + 1];
+		const branchFileKey =
```

Comment:
**issue:** @Roystbeef , we also need to update `FigmaBackfillService.buildMinimalDesignFromUrl`, which constructs `AtlassianDesign` from a given URL.

https://github.com/atlassian-labs/figma-for-jira/blob/9a59d71c167999678fd0d125bdef9366db08941a/src/infrastructure/figma/figma-backfill-service.ts#L33

We need to change:

```ts
const [, , , name] = url.pathname.split('/'); // A risky and fragile assumption that `name` always comes 4th.

const displayName = name
	? decodeURIComponent(name).replaceAll('-', ' ')
	: 'Untitled';
```

To this (not tested):

```ts
const pathComponents = url.pathname.split('/');
const keyComponentIndex = pathComponents.find(x => x === designId.fileKey);

const name = pathComponents[keyComponentIndex + 1]; // A name goes after the `fileKey` or `branchFileKey` component.
const displayName = name
	? decodeURIComponent(name).replaceAll('-', ' ')
	: 'Untitled';
```

Reply:
good flag! This code looks good to me!